Core/src/org/sleuthkit/autopsy/machinesettings/UserMachinePreferences.java

```diff
@@ -80,89 +79,56 @@
 
     private static final TempDirChoice DEFAULT_CHOICE = TempDirChoice.SYSTEM;
 
-    /**
-     * Returns the name of this computer's host name to be used as a directory
-     * in some instances.
-     *
-     * @return The name of this computer's host name to be used as a directory
-     * in some instances.
-     */
-    private static String getHostName() {
-        return NetworkUtils.getLocalHostName();
-    }
-
     /**
      * @return A subdirectory of java.io.tmpdir.
      */
     private static File getSystemTempDirFile() {
-        return Paths.get(System.getProperty("java.io.tmpdir"), AUTOPSY_SUBDIR).toFile();
-    }
-
-    /**
-     * @return A subdirectory of the open case or getSystemTempDirFile if no
-     * open case.
-     */
-    private static File getCaseTempDirFile() {
-        try {
-            Case autCase = Case.getCurrentCaseThrows();
-            String caseDirStr = autCase.getCaseDirectory();
-            switch (autCase.getCaseType()) {
-                case MULTI_USER_CASE: return Paths.get(caseDirStr, getHostName(), CASE_SUBDIR).toFile();
-                case SINGLE_USER_CASE: return Paths.get(caseDirStr, CASE_SUBDIR).toFile();
-                default: 
-                    logger.log(Level.SEVERE, "Unknown case type: " + autCase.getCaseType());
-                    return getSystemTempDirFile();
-            }
-            
-        } catch (NoCurrentCaseException ex) {
-            return getSystemTempDirFile();
-        }
+        return Paths.get(System.getProperty("java.io.tmpdir"), APP_NAME, TEMP_FOLDER_NAME).toFile();
     }
 
     /**
      * Returns the custom directory subdirectory to be used for temp files
-     * (otherwise java.io.tmpdir subdir).
+     * (otherwise null).
      *
      * @return A subdirectory of the custom user-specified path. If no path is
-     * specified, getSystemTempDirFile() is returned instead.
+     * specified, returns null.
      */
     private static File getCustomTempDirFile() {
-        String customDirectory = getCustomTempDirectory();
+        String customDirectory = getCustomTempDirectoryProperty();
         return (StringUtils.isBlank(customDirectory))
-                ? getSystemTempDirFile() : Paths.get(customDirectory, AUTOPSY_SUBDIR, getHostName()).toFile();
+                ? null : Paths.get(customDirectory, APP_NAME, TEMP_FOLDER_NAME).toFile();
     }
 
     /**
-     * Returns the temp directory file to use based on user choice.
+     * Returns the temp directory to use based on settings. This method also
+     * ensures the temp directory has been created.
+     *
+     * NOTE: The Case class also handles providing a temp directory. When
+     * altering code in this class, also look at the Case class as well.
+     * See JIRA-7505 for more information.
      *
-     * @return The directory.
+     * @return The base user-specified temporary directory.
      */
-    private static File getTempDirFile() {
+    public static String getTempDirectory() {
```

Comment:
See comments to Case class changes. I think we should make this method Path getAppTempDirectoryPath(). Note return type.